django/contrib/postgres/forms/hstore.py

```diff
@@ -23,6 +23,8 @@ def prepare_value(self, value):
     def to_python(self, value):
         if not value:
             return {}
+        if isinstance(value, dict):
+            value = json.dumps(value)
         try:
             value = json.loads(value)
         except ValueError:
```

Comment:
I guess you're forcing a dumps/loads roundtrip to make sure value is JSON serializable?